### DISCUSSION TITLE
Installing Geant4 in Multithreaded Mode

### DISCUSSION THREAD

**Discussion by ananthapadmanabhan18** [alisw/alidist#5556] [Q&A] [answered]
I wanted to install Geant4 and all the other components in Multi-threaded mode. Anyone has any idea how to proceed with it? One method i can think of is by turning the variable `GEANT4_BUILD_MULTITHREADED=ON`  in [this line](https://github.com/alisw/alidist/blob/4a737589c2c55ede833cbcad2be29dc73ee685b5/geant4.sh#L22C1-L22C37) . But I am concerned that if this will break any other dependent components like `AliRoot` etc? So, could anyone tell me How to properly install Geant4 In multi-threaded mode?

**Comment by ihrivnac** [comment, marked as answer]
To build Geant4 in Multi-threaded mode, you should add (or change) the following line in `o2-defaults.sh` (or the defaults which you use for your build):
```
env:
  GEANT4_BUILD_MULTITHREADED: 'ON'
```

This mode should not break other components (AliRoot, O2, etc.), as using Geant4 in the MT mode in AliRoot or O2 simulation requires its activation in the Geant4 configuration (`g4Config.C`)

The activation of this option would cause run time problems with both AliRoot and O2, as its support was never  done for AliRoot and it was dropped for O2 few years ago, when we moved to run-time parallelism.
